### PR TITLE
dcache : updated PoolStatisticsV0

### PR DESCRIPTION
**Pull request by jahidmixmaster**
Motivation:
New java8 time Api is Thread safe 

Modification:
Updated PoolStatisticsV0 Class .All threadLocal stores which consist of SimpleDateFormat replaced by java 8 DateTimeFormatter which is more Thread safe and no need  to create Anonymous Class

Result:
PoolStatisticsV0 Class is now up-to-date

Signed-off-by: jahid <jahid6847@gmail.com>

### CODE REVIEW

File: modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java [hunk 1]
```diff
@@ -347,19 +294,19 @@ public void getInfo(PrintWriter pw) {
     public void messageArrived(CellMessage message) {}
 
     private String getNiceDayOfCalendar(Calendar calendar) {
-        return _dayOfCalendar.get().format(calendar.getTime());
+        return calendar.getTime().toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime().format(_dayOfCalendar);
```
Comment (kofemann) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: Probably the easier version will be

```java
LocalDateTime.now().format(...);
```

File: modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java [hunk 2]
```diff
@@ -663,7 +610,7 @@ private void updateHtmlTop() throws IOException {
                 StringTokenizer st = new StringTokenizer(line);
 
                 st.nextToken();
-                String key =  _yearOfCalendar.get().format(new Date(Long.parseLong(st.nextToken())));
+                String key = new Date(Long.parseLong(st.nextToken())).toInstant().atZone(ZoneId.systemDefault()).format(_yearOfCalendar);
```
Comment (kofemann) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: what about 
```
_yearOfCalendar.format(Instant.ofEpochMilli(Long.parseLong(st.nextToken())))
```

File: modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java (hunk 1 above)
Comment (jahidmixmaster) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: i thought that is easier converting Calendar object to LocalDateTime object then call the format method

Comment (kofemann) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: Well, you get LocalDateTime from calendar
```
calendar.getTime().toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime()
```
 where I  suggest a short-cut:
```
LocalDateTime.now()
```

Comment (jahidmixmaster) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: ok. i understand what you mean but i should use calendar object which is passed get NiceDay Calendar function

File: modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java (hunk 2 above)
Comment (jahidmixmaster) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: it's not working , i got a error
 Exception in thread "main" java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: YearOfEra

Comment (anupash) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: To format an Instant a time-zone is required. Without a time-zone, the formatter does not know how to convert the instant to human date-time fields, and therefore throws an exception.

to the formatter _yearOfCalendar append 
```java
.withZone(ZoneId.systemDefault())
```

Comment (jahidmixmaster) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: yes ,i have also done simmilar like this 
new Date(Long.parseLong(st.nextToken())).toInstant().atZone(ZoneId.systemDefault()).format(_yearOfCalendar);

but tigran suggest me this
_yearOfCalendar.format(Instant.ofEpochMilli(Long.parseLong(st.nextToken())))

which is't working

Comment (jahidmixmaster) [inline modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java]: yes ,i have also done simmilar like this
```
new Date(Long.parseLong(st.nextToken())).toInstant().atZone(ZoneId.systemDefault()).format(_yearOfCalendar);
```
but tigran suggest me this
```
_yearOfCalendar.format(Instant.ofEpochMilli(Long.parseLong(st.nextToken())))
```
which is't working

**Comment by kofemann** [comment]
The contributor did not follow-up the patch. closing.

**Pull request closed without merging**